### PR TITLE
[DATA-566] Fix Flaky Test in DetectionSource

### PR DESCRIPTION
**Pull request by kharijarrett**

### CODE REVIEW

**Review by kharijarrett — COMMENTED** [review]
Hi okay so the fix here was to add another case to NextResult that times out after 2 seconds and makes the return nil. I changed the test to include a timer that checks to make sure that behavior is what's expected (time out after 2 seconds). 
I ran this 5000 times in a canon-shell-amd64 and it passed (and before the change it broke after ~1000 attempts)

File: vision/objectdetection/detection_source.go [hunk 1]
```diff
@@ -96,10 +100,19 @@ func (s *Source) Read(ctx context.Context) (image.Image, func(), error) {
...
 		return nil, nil, err
 	}
+
+	for res == nil {
+		res, err = s.NextResult(ctx)
+		if err != nil {
+			return nil, nil, err
+		}
+	}
+
```
Comment (bhaney) [inline vision/objectdetection/detection_source.go:115]: [q] Do you still need this for loop? Is it ever going to produce a nil result?

**Review by bhaney — COMMENTED** [review]
Just one question for the added for-loop, otherwise looking good

File: vision/objectdetection/detection_source.go (hunk 1 above)
Comment (kharijarrett) [reply vision/objectdetection/detection_source.go:115]: Naw I guess we don't... removing now. Gonna run the test a few thousand times again but you're right.

**Review by bhaney — APPROVED** [review]
If the flakiness doesn't come back after a few thousand test runs, then this LGTM

**Pull request merged**